server/src/main/java/org/opensearch/plugins/IdentityPlugin.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.identity.Subject;
+
+/**
+ * Plugin that provides identity and access control for OpenSearch
+ *
+ * @opensearch.experimental
+ */
+public interface IdentityPlugin {
+
+    /**
+     * Get the current subject
+     * */
+    public Subject getSubject();
```

Comment:
Could `getSubject()` return `null` vaues? If yes, we have to reconsider the API and return `Optional<Subject>` instead

Reply:
`@NotNull` on return type?